core/keyserver/paginator.go

```diff
@@ -100,37 +92,12 @@ func (s SourceMap) Next(rt *rtpb.ReadToken, lastRow *mutator.LogMessage) *rtpb.R
 	}
 }
 
-// NextShard returns the next shardID from the SourceMap.
-// Returns false if there are no more shards or shardID is not in SourceMap.
-func (s SourceMap) NextShard(shardID int64) (int64, bool) {
-	// Sorted list of shardIDs.
-	shardIDs := sortedKeys(s)
-
-	// Index of current shard.
-	i := sort.Search(len(shardIDs), func(i int) bool { return shardIDs[i] >= shardID })
-	if i == -1 {
-		// shardID isn't in SourceMap.
-		return 0, false
-	}
-	if i >= len(shardIDs)-1 {
+// NextShard returns the next shardID from the SourceList.
+// Returns false if there are no more shards or shardID is not in SourceList.
+func (s SourceList) NextShard(shardID int64) (int64, bool) {
+	if shardID >= int64(len(s))-1 {
 		// there are no more shards to iterate over.
 		return 0, false
 	}
-	return shardIDs[i+1], true
+	return shardID + 1, true
```

Comment:
Just to double check: Should this be an index in the `SourceList` slice, or should this be the ID of the queue (which, I suppose, needs to be extracted from the `MapMetadata_SourceSlice`)?

Reply:
This PR has changed the meaning of this.  
It used to be `LogID` and now it is `index of source slice`

Reply:
I'll rename this shard_index for clarity.